python/az/aro/azext_aro/custom.py

```diff
@@ -134,9 +139,55 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
                        parameters=oc)
 
 
-def aro_delete(client, resource_group_name, resource_name, no_wait=False):
+def aro_delete(cmd, client, resource_group_name, resource_name, no_wait=False):
     # TODO: clean up rbac
 
+    oc = None
+    try:
+        oc = client.get(resource_group_name, resource_name)
+    except CloudError:
+        pass
+
+    if oc:
+        master_subnet = oc.master_profile.subnet_id
+        worker_subnets = {w.subnet_id for w in oc.worker_profiles}
+
+        master_parts = parse_resource_id(master_subnet)
+        vnet = resource_id(
+            subscription=master_parts['subscription'],
+            resource_group=master_parts['resource_group'],
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=master_parts['name'],
+        )
+
+        aad = AADManager(cmd.cli_ctx)
+
+        sp_ids = []
+
+        client_id = oc.service_principal_profile.client_id
+        client_sp = aad.get_service_principal(client_id)
+        if client_sp:
+            sp_ids += client_sp.object_id
+        else:
+            logger.warning(
+                'Unable to retrieve the cluster service principal. This means '
+                'that it may have been deleted, and some cleanup may fail.'
+            )
+
+        rp_client_id = FP_CLIENT_ID
+        rp_client_sp = aad.get_service_principal(rp_client_id)
+        if rp_client_sp:
+            sp_ids += rp_client_sp.object_id
+
+        # Customers frequently remove these permissions, then cannot delete their
+        # clusters. Hence, verify this before attempting deletion.
+        for sp_id in sp_ids:
+            assign_contributor_to_vnet(cmd.cli_ctx, vnet, sp_id)
+            assign_contributor_to_routetable(cmd.cli_ctx,
+                                             worker_subnets | set(master_subnet),
```

Comment:
I think this should be:

```suggestion
                                             worker_subnets | {master_subnet},
```

`set(master_subnet)` is the set of unique chars in the string.